src/FixerDefinition/VersionSpecification.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\FixerDefinition;
+
+/**
+ * @author Andreas Möller <am@localheinz.com>
+ *
+ * @internal
+ */
+final class VersionSpecification implements VersionSpecificationInterface
+{
+    /**
+     * @var int|null
+     */
+    private $minimum;
+
+    /**
+     * @var int|null
+     */
+    private $maximum;
```

Comment:
forward support, depending on how this will turns out:
https://wiki.php.net/rfc/deprecations_php_7_2
for example. If in that RFC (or any other) features are removed from PHP which are valid now, we cannot set a max. fix level. So if `(binary)$a;` is no longer valid syntax on 7.2 or 8.0, we can longer removed that cast when one is running the tool on 7.2/8.0, but we could still do it on >7.1